src/components/dialogs/create-case-dialog.tsx

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useIntl } from 'react-intl';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { keyGenerator } from '../../utils/functions';
+import { createCase } from '../../utils/rest-api';
+import { HTTP_UNPROCESSABLE_ENTITY_STATUS } from '../../utils/UIconstants';
+import { Grid } from '@mui/material';
+import {
+    addUploadingElement,
+    removeUploadingElement,
+} from '../../redux/actions';
+import UploadNewCase from './commons/upload-new-case';
+import { ElementType } from '../../utils/elementType';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { NameCheckReturn, useNameCheck } from './commons/use-name-check';
+import { useForm } from 'react-hook-form';
+import { CASE_FILE, CASE_NAME, DESCRIPTION } from '../utils/field-constants';
+import { ErrorInput, TextInput, FieldErrorAlert } from '@gridsuite/commons-ui';
+import yup from '../utils/yup-config';
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup';
+import CustomMuiDialog from './custom-mui-dialog';
+
+const MAX_FILE_SIZE_IN_MO = 100;
+const MAX_FILE_SIZE_IN_BYTES = MAX_FILE_SIZE_IN_MO * 1024 * 1024;
+
+interface ICreateCaseDialogProps {
+    onClose: () => void;
+    open: boolean;
+}
+
+const CreateCaseDialog: React.FunctionComponent<ICreateCaseDialogProps> = ({
+    onClose,
+    open,
+}) => {
+    const intl = useIntl();
+    const dispatch = useDispatch();
+    const { snackError } = useSnackMessage();
+
+    const schema = yup.object().shape({
+        [CASE_NAME]: yup.string().required(),
+        [DESCRIPTION]: yup.string(),
+        [CASE_FILE]: yup.mixed(),
+    });
+    const createCaseFormMethods = useForm<{
+        [CASE_NAME]: string;
+        [DESCRIPTION]: string;
+        [CASE_FILE]: File | null;
+    }>({
+        mode: 'onChange',
```

Comment:
Same as CreateStudyDialog

Reply:
as discussed, if we want to keep the old behaviour, we will need form verifications 'onChange', otherwise, we will not need to use isCreationAllowed and we will remove this mode.. We need confirmation. thank you